tests/target-browser/target_browser_test.go

```diff
@@ -1624,6 +1631,278 @@ var _ = Describe("Target Browser Tests", func() {
+				args := []string{cmdGet, cmdResourceMetadata, flagKind, "", flagVersion, "", flagBackupUIDFlag, "", flagBackupPlanUIDFlag, ""}
+				args = append(args, commonArgs...)
+				args = append(args, flagName)
+				command := exec.Command(targetBrowserBinaryFilePath, args...)
+				output, err := command.CombinedOutput()
+				Expect(err).Should(HaveOccurred())
+				Expect(string(output)).Should(ContainSubstring(fmt.Sprintf("flag needs an argument: %s", flagName)))
+			})
+
+			It(fmt.Sprintf("Should fail if flag %s is given without value", cmd.GroupFlag), func() {
+				args := []string{cmdGet, cmdResourceMetadata, flagKind, "", flagName, "", flagVersion, "", flagBackupUIDFlag, "",
+					flagBackupPlanUIDFlag, ""}
+				args = append(args, commonArgs...)
+				args = append(args, flagGroup)
+				command := exec.Command(targetBrowserBinaryFilePath, args...)
+				output, err := command.CombinedOutput()
+				Expect(err).Should(HaveOccurred())
+				Expect(string(output)).Should(ContainSubstring(fmt.Sprintf("flag needs an argument: %s", flagGroup)))
+			})
+
+			It("Should fail if flags are given wrong values", func() {
+				isLast = true
+				args := []string{cmdGet, cmdResourceMetadata, flagGroup, "app", flagKind, "Deplo", flagVersion, "v",
+					flagName, "mysq", flagBackupUIDFlag, "random", flagBackupPlanUIDFlag, "random"}
+				args = append(args, commonArgs...)
+				command := exec.Command(targetBrowserBinaryFilePath, args...)
+				output, err := command.CombinedOutput()
+				Expect(err).Should(HaveOccurred())
+				Expect(string(output)).Should(ContainSubstring("did not successfully completed - 404 Not Found"))
+			})
+		})
+
+		Context("Filter Operations on absolute values for flags used", func() {
+			var (
+				backupPlanUIDs                   []string
+				noOfBackupPlansToCreate          = 1
+				noOfBackupsToCreatePerBackupPlan = 1
+				once                             sync.Once
+				isLast                           bool
+				backupUID                        string
+			)
+
+			BeforeEach(func() {
+				once.Do(func() {
+					backupUID = guid.New().String()
+					createTarget(true)
+					output, _ := exec.Command(createBackupScript, strconv.Itoa(noOfBackupPlansToCreate),
+						strconv.Itoa(noOfBackupsToCreatePerBackupPlan), "all_type_backup", backupUID, "custom", "resource-metadata").Output()
+
+					log.Info("Shell Script Output: ", string(output))
+					backupPlanUIDs = verifyBackupPlansAndBackupsOnNFS(noOfBackupPlansToCreate, noOfBackupPlansToCreate*noOfBackupsToCreatePerBackupPlan)
+					verifyBrowserCacheBPlan(noOfBackupPlansToCreate)
+				})
+			})
+
+			AfterEach(func() {
+				if isLast {
+					// delete target & remove all files & directories created for this Context - only once After all It in this context
+					deleteTarget(false)
+					for _, backupPlans := range backupPlanUIDs {
+						_, err := shell.RmRf(filepath.Join(TargetLocation, backupPlans))
+						Expect(err).To(BeNil())
+					}
```

Comment:
@vishutrilio pls use this `removeBackupPlanDir`

Reply:
Done